### PR TITLE
Additional logic for carousel v2.

### PR DESCRIPTION
**Pull request by sparhami**
- Update the current index on scroll
- Move slides around on scroll
- Implement scroll position restoration (when resetting the reference
point or when rotating)
- Have responsive attributes logic ignore attributes that have no configuration option (e.g. style)
- Make media query attribute parsing logic more robust

### CODE REVIEW

File: extensions/amp-carousel/0.2/carousel.js [hunk 1]
```diff
@@ -569,7 +577,7 @@ export class Carousel {
    * @private
    */
   resetSlideTransforms_() {
-    // TODO(sparhami) implement
+    this.slides_.forEach(slide => this.setElementTransform_(slide, 0, 0));
```
Comment (alanorozco) [inline extensions/amp-carousel/0.2/carousel.js]: nit: would be nice to annotate what `0, 0` are.

File: extensions/amp-carousel/0.2/responsive-attributes.js [hunk 2]
```diff
@@ -89,19 +94,26 @@ export class ResponsiveAttributes {
...
+          // The media query is everything before the value.
+          const mediaQuery = part.slice(0, index).trim();
+          const mediaQueryList = window.matchMedia(mediaQuery);
+
+          return {
+            mediaQueryList,
+            value,
+          };
+        })
+        .filter(item => item);
```
Comment (alanorozco) [inline extensions/amp-carousel/0.2/responsive-attributes.js]: Superfluous filter?

Comment (sparhami) [reply extensions/amp-carousel/0.2/responsive-attributes.js]: Added a comment on what this is doing.

File: extensions/amp-carousel/0.2/carousel.js (hunk 1 above)
Comment (sparhami) [reply extensions/amp-carousel/0.2/carousel.js]: I changed the code to be more clear here.

File: extensions/amp-carousel/0.2/responsive-attributes.js (hunk 2 above)
Comment (sparhami) [reply extensions/amp-carousel/0.2/responsive-attributes.js]: Added a comment to the code explaining why it is there.

**Pull request merged**